File: python/raindrops/raindrops.py
def convert(number):
    factors = {
        3: "Pling",
        5: "Plang",
        7: "Plong"
    }

    rval = []
    for factor, msg in factors.items():
        if (number % factor) == 0:
            rval.append(msg)

    return "".join(rval) or str(number)

File: go/raindrops/raindrops.go
package raindrops

import "fmt"

var numTable = map[int]string{
	3: "Pling",
	5: "Plang",
	7: "Plong",
}

func Convert(number int) string {

	var result string

	for k, v := range numTable {
		if number%k == 0 {
			result += v
		}
	}

	if len(result) == 0 {
		result = fmt.Sprintf("%d", number)
	}

	return result
}
